core/src/main/java/io/temporal/samples/hello/HelloAccumulator.java

```diff
@@ -0,0 +1,625 @@
+    }
+
+    public Greeting() {}
+
+    @Override
+    public String toString() {
+      return "Greeting [greetingText="
+          + greetingText
+          + ", bucket="
+          + bucket
+          + ", greetingKey="
+          + greetingKey
+          + "]";
+    }
+  }
+
+  /**
+   * The Workflow Definition's Interface must contain one method annotated with @WorkflowMethod.
+   *
+   * <p>Workflow Definitions should not contain any heavyweight computations, non-deterministic
+   * code, network calls, database operations, etc. Those things should be handled by the
+   * Activities.
+   *
+   * @see io.temporal.workflow.WorkflowInterface
+   * @see io.temporal.workflow.WorkflowMethod
+   */
+  @WorkflowInterface
+  public interface AccumulatorWorkflow {
+    /**
+     * This is the method that is executed when the Workflow Execution is started. The Workflow
+     * Execution completes when this method finishes execution.
+     */
+    @WorkflowMethod
+    String accumulateGreetings(
+        String bucketKey, ArrayDeque<Greeting> greetings, Set<String> allGreetingsSet);
+
+    // Define the workflow waitForName signal method. This method is executed when
+    // the workflow receives a greeting signal.
+    @SignalMethod
+    void waitforGreeting(Greeting greeting);
+
+    // Define the workflow exit signal method. This method is executed when the
+    // workflow receives an exit signal.
+    @SignalMethod
+    void exit();
+  }
+
+  /**
+   * This is the Activity Definition's Interface. Activities are building blocks of any Temporal
+   * Workflow and contain any business logic that could perform long running computation, network
+   * calls, etc.
+   *
+   * <p>Annotating Activity Definition methods with @ActivityMethod is optional.
+   *
+   * @see io.temporal.activity.ActivityInterface
+   * @see io.temporal.activity.ActivityMethod
+   */
+  @ActivityInterface
+  public interface GreetingActivities {
+    String composeGreeting(ArrayDeque<Greeting> greetings);
+  }
+
+  /** Simple activity implementation. */
+  static class GreetingActivitiesImpl implements GreetingActivities {
+
+    // here is where we process all of the signals together
+    @Override
+    public String composeGreeting(ArrayDeque<Greeting> greetings) {
+      List<String> greetingList =
+          greetings.stream().map(u -> u.greetingText).collect(Collectors.toList());
+      return "Hello (" + greetingList.size() + ") robots: " + greetingList + "!";
+    }
+  }
+
+  // Main workflow method
+  public static class AccumulatorWorkflowImpl implements AccumulatorWorkflow {
+
+    private final GreetingActivities activities =
+        Workflow.newActivityStub(
+            GreetingActivities.class,
+            ActivityOptions.newBuilder().setStartToCloseTimeout(Duration.ofSeconds(2)).build());
+
+    private static final Logger logger = LoggerFactory.getLogger(AccumulatorWorkflowImpl.class);
+    String bucketKey;
+    ArrayDeque<Greeting> greetings;
```

Comment:
I think doing this will break the tests in HelloAccumulatorTest. Most of the other samples have their Impl as public. 
Setting Greeting as private also breaks the test.